Sources/X509/RDNAttribute.swift

```diff
@@ -26,67 +26,134 @@ extension RelativeDistinguishedName {
     /// of the node will be an `ASN1PrintableString` or `ASN1UTF8String`, but the value can only
     /// be derived by inspection.
     public struct Attribute {
+        public struct Value: Hashable, Sendable {
+            @usableFromInline
+            enum Storage: Sendable {
+                /// ``ASN1PrintableString``
+                case printable(String)
+                /// ``ASN1UTF8String``
+                case utf8(String)
+                /// `.any` can still contain bytes which are equal to the DER representation of `.printable` or `.utf8`
+                /// the custom `Hashable` conformance takes care of this and treats them as equal.
+                case any(ASN1Any)
+            }
+            
+            @usableFromInline
+            var storage: Storage
+            
+            @inlinable
+            init(storage: Storage) {
+                self.storage = storage
+            }
+        }
         /// The type of this attribute.
         ///
         /// Common types can be found in `ASN1ObjectIdentifier.RDNAttributeType`.
         public var type: ASN1ObjectIdentifier
 
-        // TODO(cory): This feels gross, we should have a better representation that reduces the runtime type checking and recomputation.
-        // These are all strings in other places, but we can't necessarily guarantee that. Probably want a computed property.
         /// The value of this attribute.
-        public var value: ASN1Any
+        public var value: Attribute.Value
 
         /// Create a new attribute from a given type and value.
         ///
         /// - Parameter type: The type of the attribute.
-        /// - Parameter value: The value of the attribute, wrapped in an `ASN1Any`.
+        /// - Parameter value: The value of the attribute.
         @inlinable
-        public init(type: ASN1ObjectIdentifier, value: ASN1Any) {
+        public init(type: ASN1ObjectIdentifier, value: Attribute.Value) {
```

Comment:
there a three public inits for `RelativeDistinguishedName.Attribute.Value` defined in line 115 - 136.

Reply:
Why did you decide to put them into a separate extension? (Just curious because it made them quite hidden in the review)

Reply:
I always try to put the symbols of nested types into extensions to reduce nesting. I only left the `init(storage: Storage)` in there because it needs to be there or it doesn't suppress the auto syntezied member wise init (which we can't use because it isn't `@inlinable`)